src/backend/updating-llvm.md

```diff
@@ -19,18 +19,13 @@ policy!), but for now these are rough guidelines!
 
 ## Why update LLVM?
 
-There are a few reasons nowadays that we want to update LLVM in one way or
-another:
+There are two reasons we would want to update LLVM:
 
 * A bug could have been fixed! Often we find bugs in the compiler and fix
   them upstream in LLVM. We'll want to pull fixes back to the compiler itself as
   they're merged upstream.
 
-* A new feature may be available in LLVM that we want to use in rustc,
-  but we don't want to wait for a full LLVM release to test it out.
-
-* LLVM itself may have a new release and we'd like to update to this LLVM
-  release.
+* LLVM itself may have a new release.
```

Comment:
These two things are different, a new feature that isn't released may be cherry-picked into Rust's LLVM fork, e.g. https://github.com/rust-lang/llvm-project/pull/135.

Reply:
interesting, because @nikic [said it is not done any more](https://rust-lang.zulipchat.com/#narrow/stream/187780-t-compiler.2Fwg-llvm/topic/is.20Updating.20LLVM.20on.20rustc-dev-guide.20current/near/292708097), maybe because it is rare in the recent past

Reply:
@nikic Is that true? I thought the above PR is an example of a new feature cherry-picked into our LLVM fork.

Reply:
The relevant distinction here is between a cherry-pick (bug fix or self-contained feature) and a rebase (new release). The thing we don't do any more is to rebase without a new (pending) release.

Reply:
Thanks for clarifying! I think that should be documented above, the current sounds somewhat confusing.